extensions/eda/plugins/event_source/url_check.py

```diff
@@ -26,30 +27,46 @@
 
 OK = 200
 
-
 async def main(queue: asyncio.Queue, args: dict[str, Any]) -> None:
     """Poll a set of URLs and send events with status."""
     urls = args.get("urls", [])
     delay = int(args.get("delay", 1))
     verify_ssl = args.get("verify_ssl", True)
 
+    # Added initialization for client_error to handle cases where an exception occurs
+    client_error = ""
+
     if not urls:
         return
 
+    # Added url_states dictionary to track the last known status of each URL
+    url_states = {url: None for url in urls}
+
     while True:
         try:
             async with aiohttp.ClientSession() as session:
                 for url in urls:
-                    async with session.get(url, verify_ssl=verify_ssl) as resp:
-                        await queue.put(
-                            {
-                                "url_check": {
-                                    "url": url,
-                                    "status": "up" if resp.status == OK else "down",
-                                    "status_code": resp.status,
-                                },
+                    try:
+                        async with session.get(url, ssl=verify_ssl) as resp:
+                            status = "up" if resp.status == OK else "down"
+                            status_code = resp.status
+                    except aiohttp.ClientError as exc:
```

Comment:
I think to set the status to "down" is not correct, since the ClientError can be raised by a variety of reasons that doesn't necessarily mean the url is down, like for example networking issues in the client side. This would be something that the plugin is already doing wrong in the upper try/except block. 

Instead, I suggest a new option with "false" as default named for example "send_error_events" and send the ClientError as event if its set. 

thoughts? @bzwei